doc/rule_flags.md

```diff
@@ -0,0 +1,393 @@
+      "rules": [
+        {
+          "type": "match_list",
+          "property": "host_name",
+          "values": [
+            "srv_123",
+            "srv_456"
+          ],
+          "on_match": "off",
+          "on_miss": "continue"
+        },
+        {
+          "type": "match_list",
+          "property": "host_name",
+          "values": [
+            "srv_789"
+          ],
+          "on_match": "on",
+          "on_miss": "continue"
+        },
+        {
+          "type": "sample",
+          "rate": 0.01,
+          "properties": ["cluster", "db"],
+          "on_match": "on",
+          "on_miss": "off"
+        }
+      ]
+    },
+  ],
+  "updated": 1519247256.0626957
+}
+```
+
+## Examples
+
+Here are some common use cases, and how to implement them with rules:
+
+### Always off
+
+```
+{
+	"name": "test.off",
+	"active": false,
+	"rules": []
+}
+```
+
+### Always on
+
+```
+{
+	"name": "test.on",
+	"active": true,
+	"rules": [
+		{
+			"type": "sample",
+			"properties": [],
+			"rate": 1,
+			"on_match": "on",
+			"on_miss": "off"
+		}
+	]
+}
+```
+
+### Simple sampling
+
+On 1% of the time:
+
+```
+{
+	"name": "test.random",
+	"active": true,
+	"rules": [
+		{
+			"type": "sample",
+			"properties": [],
+			"rate": 0.01,
+			"on_match": "on",
+			"on_miss": "off"
+		}
+	]
+}
+```
+
+### Sampling by property
+
+On for 1% of users. Consistently on for the same set of users.
+
+```
+{
+	"name": "test.random_by",
+	"active": true,
+	"rules": [
+		{
+			"type": "sample",
+			"properties": ["user"],
+			"rate": 0.01,
+			"on_match": "on",
+			"on_miss": "off"
+		}
+	]
+}
+```
+
+### Whitelist
+
+On for only Alice and Bob:
+
+```
+{
+	"name": "test.whitelist",
+	"active": true,
+	"rules": [
+		{
+			"type": "match_list",
+			"property": "user",
+			"values": ["alice", "bob"],
+			"on_match": "on",
+			"on_miss": "off"
+		}
+	]
+}
+```
+
+### Blacklist
+
+On for everyone except Xavier:
+
+```
+{
+	"name": "test.blacklist",
+	"active": true,
+	"rules": [
+		{
+			"type": "match_list",
+			"property": "user",
+			"values": ["xavier"],
+			"on_match": "off",
+			"on_miss": "on"
+		}
+	]
+}
+```
+
+### Whitelist some, sample from the rest
+
+This is useful for a new feature, with some explicit test users, and a random selection of other users:
+
+
+```
+{
+	"name": "test.random_by_with_whitelist",
+	"active": true,
+	"rules": [
+		{
+			"type": "match_list",
+			"property": "user",
+			"values": ["test_user1", "test_user2"],
+			"on_match": "on",
+			"on_miss": "continue"
+		},
+		{
+			"type": "sample",
+			"properties": ["user"],
+			"rate": 0.01,
+			"on_match": "on",
+			"on_miss": "off"
+		}
+	]
+}
+```
+
+### Sample from only certain users
+
+Only Alice should have this feature, and she should only see it for 5% of requests:
+
+
+```
+{
+	"name": "test.whitelist_then_sample",
+	"active": true,
+	"rules": [
+		{
+			"type": "match_list",
+			"property": "user",
+			"values": ["alice"],
+			"on_match": "on",
```

Comment:
on_match should be continue here and on_miss should be off?

Reply:
Nice catch!